firedrake/plot.py

```diff
@@ -3,110 +3,414 @@
     if len(functions) == 0:
         return None
+    import os
+    from firedrake import __file__ as firedrake__file__
     figure, ax = plt.subplots()
-    func_axis = plt.axes([0.25, 0.025, 0.65, 0.03], axisbg='lightgoldenrodyellow')
-    func_slider = Slider(func_axis, "Func Select", 0.1, len(functions), valinit=0)
+    func_axis = plt.axes([0.3, 0.025, 0.65, 0.03],
+                         axisbg='lightgoldenrodyellow')
+    func_slider = Slider(func_axis, "Func Select",
+                         0.1, len(functions), valinit=0)
     func_slider.valtext.set_text('0')
+    play_axis = plt.axes([0.05, 0.025, 0.1, 0.03])
+    play_image = mpimg.imread(os.path.join(
+        os.path.dirname(firedrake__file__), 'icons/play.png'))
+    play_button = Button(play_axis, "", image=play_image)
+    play_axis._button = play_button  # Hacking: keep a reference of button
+    plus_axis = plt.axes([0.08, 0.025, 0.1, 0.03])
+    plus_image = mpimg.imread(os.path.join(
+        os.path.dirname(firedrake__file__), 'icons/plus.png'))
+    plus_button = Button(plus_axis, "", image=plus_image)
+    plus_axis._button = plus_button  # Hacking: keep a reference of button
+    minus_axis = plt.axes([0.02, 0.025, 0.1, 0.03])
+    minus_image = mpimg.imread(os.path.join(
+        os.path.dirname(firedrake__file__), 'icons/minus.png'))
+    minus_button = Button(minus_axis, "", image=minus_image)
+    minus_axis._button = minus_button  # Hacking: keep a reference of button
+
+    class Player:
+        STOP = 0
+        PLAYING = 1
+        CLOSED = -1
+        status = STOP
+        frame_interval = 0.5
+    player = Player()  # Use array to allow its value to be changed
+
+    def handle_close(event):
+        player.status = Player.CLOSED
+    figure.canvas.mpl_connect('close_event', handle_close)
 
     def update(val):
         val = int(val - 0.1)
         func_slider.valtext.set_text('{:.0f}'.format(val))
         ax.cla()
-        plot(functions[val], ax, **kwargs)
+        plot(functions[val], num_points, ax, **kwargs)
+        plt.pause(0.01)
     update(0)
     func_slider.on_changed(update)
+
+    def auto_play(event):
+        if player.status == Player.PLAYING:
+            player.status = Player.STOP
+            return
+        curr = 0
+        player.status = Player.PLAYING
+        while curr < len(functions) and player.status == Player.PLAYING:
+            curr += 1
+            func_slider.set_val(curr)
+            plt.pause(player.frame_interval)
+        player.status = Player.STOP
+    play_button.on_clicked(auto_play)
+
+    def increase_speed(event):
+        player.frame_interval /= 2
+    plus_button.on_clicked(increase_speed)
+
+    def decrease_speed(event):
+        player.frame_interval *= 2
+    minus_button.on_clicked(decrease_speed)
+
     return figure
 
 
-def plot(function, axes=None, num_points=100, **kwargs):
+def plot(function,
+         num_sample_points=100,
```

Comment:
Or perhaps default to `None` and pick a reasonable default based on the degree of the element passed in.